File: n1/1.rb
# Проверка простого числа
def smpl(n)
  return false if n < 2
  (2..Math.sqrt(n)).each { |i| return false if n % i == 0 }
  true
end
print "Введите число для определения разности между ним и ближайшим простым: "
N = gets.to_i
if N > 2
  # Поиск ближайшего простого числа <= N
  smaller = N
  smaller -= 1 until smpl(smaller)
  
  # Поиск ближайшего простого числа >= N
  larger = N
  larger += 1 until smpl(larger)
  
  # Вывод разности
  puts "Разность: #{[(N - smaller), (larger - N)].min}"
else
  puts "Разность: #{N - 2}"
end

File: n1/1.go
package main

import (
    "fmt"
    "math"
)

// Функция для проверки простого числа
func smpl(n int) bool {
	if n < 2 {
        	return false
    	}
    	for i := 2; i <= int(math.Sqrt(float64(n))); i++ {
        	if n%i == 0 {
            	return false
        	}
    	}
    	return true
}

func main() {
    	var N int
    	fmt.Print("Введите число для определения разности между ним и ближайшим простым: ")
    	fmt.Scan(&N)
	if (N > 2){
	    	// Поиск ближайшего простого числа, меньшего или равного N
	    	smaller := N
	    	for !smpl(smaller) {
	        	smaller--
	    	}
	
	    	// Поиск ближайшего простого числа, большего или равного N
	    	larger := N
	    	for !smpl(larger) {
	        	larger++
	    	}
	
	    	// Сравнение расстояний и вывод результата
	    	if N-smaller <= larger-N {
	        	fmt.Println("Разность:", N-smaller)
	    	} else {
	        	fmt.Println("Разность:", larger-N)
	    	}
	}
	fmt.Println(N - 2)
}
